File: spring-webflux/src/main/java/software/darkmatter/school/blog/domain/post/data/Post.java
package software.darkmatter.school.blog.domain.post.data;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.relational.core.mapping.Table;
import software.darkmatter.school.blog.domain.user.data.User;

import java.time.OffsetDateTime;

@Data
@Table("posts")
public class Post {
    @Id
    private Long id;

    private String title;

    private String summary;

    private String content;

    private OffsetDateTime createdAt;

    private Long createdByUserId;

    @Transient
    private User createdBy;

    private OffsetDateTime updatedAt;

    private Long updatedByUserId;

    private OffsetDateTime deletedAt;

    private Long deletedByUserId;

    private OffsetDateTime publishedAt;
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/post/data/Post.kt
package software.darkmatter.school.blog.domain.post.data

import org.springframework.data.annotation.Id
import org.springframework.data.annotation.Transient
import org.springframework.data.relational.core.mapping.Table
import software.darkmatter.school.blog.domain.user.domain.User
import java.time.OffsetDateTime

@Table("posts")
data class Post(
    @Id
    val id: Long? = null,
    var title: String?,
    var summary: String?,
    var content: String?,
    val createdAt: OffsetDateTime,
    val createdByUserId: Long,
    var updatedAt: OffsetDateTime,
    var updatedByUserId: Long,
    var deletedAt: OffsetDateTime?,
    var deletedByUserId: Long?,
    var publishedAt: OffsetDateTime?,
) {

    @Transient
    var createdBy: User? = null
}
